### ISSUE TITLE
ALTDEUS: Beyond Chronos - IL2CPP chainloader fails to execute

### ISSUE THREAD

**Issue by Aarc-Noa** [BepInEx/BepInEx#287]
### What happened?

No change - the chainloader fails to execute. This seemingly does not affect the game in any major way.

### Steps to reproduce

1. Install BepInEx 6 IL2CPP x64 on ALTDEUS: Beyond Chronos (Steam version 1.0.1)
2. Start the game
3. Check the logs (console if enabled, or log file)
4. Notice the IL2CPP chainloader failing to execute

### BepInEx Distribution

Bleeding Edge from BepisBuilds

### Log outputs

```txt
[Message:   BepInEx] Chainloader initialized
[Fatal  :   BepInEx] Unable to execute IL2CPP chainloader
[Error  :   BepInEx] System.TypeLoadException: Value type instance size (108405456) cannot be zero, negative, or bigger than 1Mb
  at UnityEngine.Application+LogCallback.op_Implicit (System.Action`3[T1,T2,T3] ) [0x00000] in <5b78ab0f4ea14994886fa3e0ad338498>:0 
  at BepInEx.IL2CPP.Logging.IL2CPPUnityLogSource..ctor () [0x00011] in <08c28cfe3f9143da91029cbb9864e38a>:0 
  at BepInEx.IL2CPP.IL2CPPChainloader.OnInvokeMethod (System.IntPtr method, System.IntPtr obj, System.IntPtr parameters, System.IntPtr exc) [0x0002a] in <08c28cfe3f9143da91029cbb9864e38a>:0 
[Fatal  :   BepInEx] Unable to execute IL2CPP chainloader
[Error  :   BepInEx] System.TypeLoadException: Failure has occurred while loading a type.
  at UnityEngine.Application+LogCallback.op_Implicit (System.Action`3[T1,T2,T3] ) [0x00000] in <5b78ab0f4ea14994886fa3e0ad338498>:0 
  at BepInEx.IL2CPP.Logging.IL2CPPUnityLogSource..ctor () [0x00011] in <08c28cfe3f9143da91029cbb9864e38a>:0 
  at BepInEx.IL2CPP.IL2CPPChainloader.OnInvokeMethod (System.IntPtr method, System.IntPtr obj, System.IntPtr parameters, System.IntPtr exc) [0x0002a] in <08c28cfe3f9143da91029cbb9864e38a>:0
```

### Environment

```markdown
- OS: Windows 11 (Build 22000.100) - Similarly happened in Windows 10 20H2 and 21H1
- BepInEx: IL2CPP x64 6.0.0.401 (Bleeding Edge)
- Game: ALTDEUS: Beyond Chronos (Steam, version 1.0.1) - IL2CPP, x64, Unity 2019.4.16
```

**Comment by bbepis** [comment]
Hi,
Are you able to try disabling `[Logging].UnityLogListening` in `BepInEx.cfg`?

**Comment by Aarc-Noa** [comment]
It seems that this might resolve the issue: 
```
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Info   :   BepInEx] 0 plugins to load
[Message:   BepInEx] Chainloader startup complete
```

As attempting to load [UnityExplorer](github.com/sinai-dev/UnityExplorer/releases/tag/4.2.0) as "test mod" also fails on something log related, I'm going to assume this build of the game has something against captuing Unity logging.

**Comment by bbepis** [comment]
It's not necessarily the case that the game is actively trying to prevent logging, but it's more likely that the methods around it are stripped out.
Unhollower provides a way to unstrip the unity assemblies which you can try out. You need to download the [unity mono runtime libraries](https://github.com/LavaGang/Unity-Runtime-Libraries/raw/master/2019.4.16.zip) for this version of unity, and extract them into `BepInEx/unity-libs`

After doing that you'll need to delete your `BepInEx/unhollowed` directory and launch the game again

This is normally done automatically, but it seems like the regular place we download base libraries from doesn't have this version

**Comment by Aarc-Noa** [comment]
Yeah I didn't mean to imply that the game actively seeked to prevent logging, my mistake.

As for Unhollower, it seems that it did exactly that (download and attempt unstripping), but there were some methods and fields that failed to be restored: 

```
[Message: Preloader] BepInEx 6.0.0-be.401 - ALTDEUS
[Message: Preloader] BLEEDING EDGE Build #401 from 9cc0211a780ba43cf2565c255e272f8264a6b80e at master
[Info   :   BepInEx] System platform: Bits64, Windows
[Info   : Preloader] Running under Unity v2019.4.16
[Message: Preloader] Downloading unity base libraries
[Message: Preloader] Extracting downloaded unity base libraries
[Message: Preloader] Generating Il2CppUnhollower assemblies
[Info   : Preloader] Generating Il2CppDumper intermediate assemblies
[Info   : Preloader] Executing Il2CppUnhollower generator
[Info   :Unhollower] Reading assemblies...
[Info   :Unhollower] Done in 00:00:00.0060303
[Info   :Unhollower] Reading system assemblies...
[Info   :Unhollower] Done in 00:00:00.0349010
[Info   :Unhollower] Reading unity assemblies...
[Info   :Unhollower] Done in 00:00:01.7789952
[Info   :Unhollower] Creating rewrite assemblies...
[Info   :Unhollower] Done in 00:00:00.0062938
[Info   :Unhollower] Computing renames...
[Info   :Unhollower] Done in 00:00:00.0344576
[Info   :Unhollower] Creating typedefs...
[Info   :Unhollower] Done in 00:00:00.0425020
[Info   :Unhollower] Computing struct blittability...
[Info   :Unhollower] Done in 00:00:00.0033470
[Info   :Unhollower] Filling typedefs...
[Info   :Unhollower] Done in 00:00:00.0209238
[Info   :Unhollower] Filling generic constraints...
[Info   :Unhollower] Done in 00:00:00.0016169
[Info   :Unhollower] Creating members...
[Info   :Unhollower] Done in 00:00:00.3449577
[Info   :Unhollower] Scanning method cross-references...
[Info   :Unhollower] Done in 00:00:00.8694826
[Info   :Unhollower] Finalizing method declarations...
[Info   :Unhollower] Done in 00:00:00.7047781
[Info   :Unhollower] 0 total potentially dead methods
[Info   :Unhollower] Filling method parameters...
[Info   :Unhollower] Done in 00:00:00.1220764
[Info   :Unhollower] Creating static constructors...
[Info   :Unhollower] Done in 00:00:00.2334829
[Info   :Unhollower] Creating value type fields...
[Info   :Unhollower] Done in 00:00:00.0517500
[Info   :Unhollower] Creating enums...
[Info   :Unhollower] Done in 00:00:00.0154027
[Info   :Unhollower] Creating IntPtr constructors...
[Info   :Unhollower] Done in 00:00:00.0217529
[Info   :Unhollower] Creating type getters...
[Info   :Unhollower] Done in 00:00:00.0478022
[Info   :Unhollower] Creating non-blittable struct constructors...
[Info   :Unhollower] Done in 00:00:00.0127300
[Info   :Unhollower] Creating generic method static constructors...
[Info   :Unhollower] Done in 00:00:00.0252975
[Info   :Unhollower] Creating field accessors...
[Info   :Unhollower] Done in 00:00:00.6402012
[Info   :Unhollower] Filling methods...
[Info   :Unhollower] Done in 00:00:01.5539852
[Info   :Unhollower] Generating implicit conversions...
[Info   :Unhollower] Done in 00:00:00.0153704
[Info   :Unhollower] Creating properties...
[Info   :Unhollower] Done in 00:00:00.0301862
[Info   :Unhollower] Unstripping types...
[Info   :Unhollower] Done in 00:00:00.0269789
[Info   :Unhollower] Unstripping fields...
[Info   :Unhollower]
[Info   :Unhollower] 521 fields restored
[Info   :Unhollower] 9 fields failed to restore
[Info   :Unhollower] Done in 00:00:00.0190264
[Info   :Unhollower] Unstripping methods...
[Info   :Unhollower]
[Info   :Unhollower] 13606 methods restored
[Info   :Unhollower] 873 methods failed to restore
[Info   :Unhollower] Done in 00:00:00.4690761
[Info   :Unhollower] Unstripping method bodies...
[Info   :Unhollower]
[Info   :Unhollower] IL unstrip statistics: 6602 successful, 2391 failed
[Info   :Unhollower] Done in 00:00:00.4098934
[Info   :Unhollower] Generating forwarded types...
[Info   :Unhollower] Done in 00:00:00.0116402
[Info   :Unhollower] Writing xref cache...
[Info   :Unhollower] Done in 00:00:00.3516157
[Info   :Unhollower] Writing assemblies...
[Info   :Unhollower] Done in 00:00:01.2954840
[Info   :Unhollower] Writing method pointer map...
[Info   :Unhollower] Done in 00:00:00.0276807
[Info   :Unhollower] Done!
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Info   :   BepInEx] 0 plugins to load
[Message:   BepInEx] Chainloader startup complete
```

**Comment by bbepis** [comment]
It's expected for a sizable amount of methods to fail being unstripped. Does re-enabling unity logging still cause the same issue?

**Comment by Aarc-Noa** [comment]
The issue mentioned in OP does return, yes.
```
[Message: Preloader] BepInEx 6.0.0-be.401 - ALTDEUS
[Message: Preloader] BLEEDING EDGE Build #401 from 9cc0211a780ba43cf2565c255e272f8264a6b80e at master
[Info   :   BepInEx] System platform: Bits64, Windows
[Info   : Preloader] Running under Unity v2019.4.16
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Fatal  :   BepInEx] Unable to execute IL2CPP chainloader
[Error  :   BepInEx] System.TypeLoadException: Value type instance size (108405456) cannot be zero, negative, or bigger than 1Mb
  at UnityEngine.Application+LogCallback.op_Implicit (System.Action`3[T1,T2,T3] ) [0x00000] in <7e7f12a1ac424871a54ba8ee6d0f4ca7>:0
  at BepInEx.IL2CPP.Logging.IL2CPPUnityLogSource..ctor () [0x00011] in <08c28cfe3f9143da91029cbb9864e38a>:0
  at BepInEx.IL2CPP.IL2CPPChainloader.OnInvokeMethod (System.IntPtr method, System.IntPtr obj, System.IntPtr parameters, System.IntPtr exc) [0x0002a] in <08c28cfe3f9143da91029cbb9864e38a>:0
[Fatal  :   BepInEx] Unable to execute IL2CPP chainloader
[Error  :   BepInEx] System.TypeLoadException: Failure has occurred while loading a type.
  at UnityEngine.Application+LogCallback.op_Implicit (System.Action`3[T1,T2,T3] ) [0x00000] in <7e7f12a1ac424871a54ba8ee6d0f4ca7>:0
  at BepInEx.IL2CPP.Logging.IL2CPPUnityLogSource..ctor () [0x00011] in <08c28cfe3f9143da91029cbb9864e38a>:0
  at BepInEx.IL2CPP.IL2CPPChainloader.OnInvokeMethod (System.IntPtr method, System.IntPtr obj, System.IntPtr parameters, System.IntPtr exc) [0x0002a] in <08c28cfe3f9143da91029cbb9864e38a>:0
```

**Comment by bbepis** [comment]
Unfortunately this seems to be an issue within [Il2cppAssemblyUnhollower](https://github.com/knah/Il2CppAssemblyUnhollower) which is what we use to interact with the IL2CPP domain. Specifically each unity version can differ in how it's method structs are laid out and a mismatch can cause weird issues like what you're seeing, so it's very likely that a handler for 2019.4 will need to be made.

The main developer is currently rewriting the tool to better accommodate different unity versions so he probably wouldn't be able to look into it right away, but if I have some free time I could possibly do it myself.

**Comment by plasmatank** [comment]
Basically the same error
`[Message: Preloader] BepInEx 6.0.0-be.568 - Touhou Blooming Soul
[Message: Preloader] BLEEDING EDGE Build #568 from e4c9ae411502522e6495a601b02f1d872b4196c5 at master
[Info   :   BepInEx] System platform: Windows 10 64-bit
[Info   :   BepInEx] Process bitness: 64-bit (x64)
[Info   : Preloader] Running under Unity v2021.2.6
[Message: Preloader] Downloading unity base libraries
[Message: Preloader] Extracting downloaded unity base libraries
[Message: Preloader] Generating Il2CppUnhollower assemblies
[Info   :    Cpp2IL] [Library] Initializing Metadata...
[Info   :    Cpp2IL] [Library] Using actual IL2CPP Metadata version 29
[Info   :    Cpp2IL] [Library] Initialized Metadata in 823ms
[Info   :    Cpp2IL] [Library] Searching Binary for Required Data...
[Info   :    Cpp2IL] [Library] Got Binary codereg: 0x181D2AF30, metareg: 0x182048200 in 164ms.
[Info   :    Cpp2IL] [Library] Initializing Binary...
[Info   :    Cpp2IL] [Library] Initialized Binary in 752ms
[Info   :    Cpp2IL] [Library] Mapping pointers to Il2CppMethodDefinitions...
[Info   :    Cpp2IL] [Library] Processed 90096 OK (103ms)
[Info   :    Cpp2IL] [Program] Building assemblies...This may take some time.
[Info   :    Cpp2IL] [Program] Finished Building Assemblies in 1323ms
[Info   :    Cpp2IL] [Program] Fixing up explicit overrides. Any warnings you see here aren't errors - they usually indicate improperly stripped or obfuscated types, but this is not a big deal. This should only take a second...
[Info   :    Cpp2IL] [Program] Fixup complete (207ms)
[Info   :    Cpp2IL] Total time: 00:00:03.5870177
[Info   : Preloader] Executing Il2CppUnhollower generator
[Info   :Unhollower] Reading assemblies...
[Info   :Unhollower] Done in 00:00:00.0032346
[Info   :Unhollower] Reading system assemblies...
[Info   :Unhollower] Done in 00:00:00.0228338
[Info   :Unhollower] Reading unity assemblies...
[Info   :Unhollower] Done in 00:00:00.0305602
[Info   :Unhollower] Creating rewrite assemblies...
[Info   :Unhollower] Done in 00:00:00.0047722
[Info   :Unhollower] Computing renames...
[Info   :Unhollower] Done in 00:00:00.0167373
[Info   :Unhollower] Creating typedefs...
[Info   :Unhollower] Done in 00:00:00.0250811
[Info   :Unhollower] Computing struct blittability...
[Info   :Unhollower] Done in 00:00:00.0022962
[Info   :Unhollower] Filling typedefs...
[Info   :Unhollower] Done in 00:00:00.0093291
[Info   :Unhollower] Filling generic constraints...
[Info   :Unhollower] Done in 00:00:00.0006570
[Info   :Unhollower] Creating members...
[Info   :Unhollower] Done in 00:00:00.2295572
[Info   :Unhollower] Scanning method cross-references...
[Info   :Unhollower] Done in 00:00:00.3162036
[Info   :Unhollower] Finalizing method declarations...
[Info   :Unhollower] Done in 00:00:00.5870919
[Info   :Unhollower] 156 total potentially dead methods
[Info   :Unhollower] Filling method parameters...
[Info   :Unhollower] Done in 00:00:00.1383634
[Info   :Unhollower] Creating static constructors...
[Info   :Unhollower] Done in 00:00:00.1569565
[Info   :Unhollower] Creating value type fields...
[Info   :Unhollower] Done in 00:00:00.0404219
[Info   :Unhollower] Creating enums...
[Info   :Unhollower] Done in 00:00:00.0122509
[Info   :Unhollower] Creating IntPtr constructors...
[Info   :Unhollower] Done in 00:00:00.0154403
[Info   :Unhollower] Creating type getters...
[Info   :Unhollower] Done in 00:00:00.0243499
[Info   :Unhollower] Creating non-blittable struct constructors...
[Info   :Unhollower] Done in 00:00:00.0030089
[Info   :Unhollower] Creating generic method static constructors...
[Info   :Unhollower] Done in 00:00:00.0155117
[Info   :Unhollower] Creating field accessors...
[Info   :Unhollower] Done in 00:00:00.2879333
[Info   :Unhollower] Filling methods...
[Info   :Unhollower] Done in 00:00:00.7886715
[Info   :Unhollower] Generating implicit conversions...
[Info   :Unhollower] Done in 00:00:00.0081187
[Info   :Unhollower] Creating properties...
[Info   :Unhollower] Done in 00:00:00.0271470
[Info   :Unhollower] Unstripping types...
[Info   :Unhollower] Done in 00:00:00.0162414
[Info   :Unhollower] Unstripping fields...
[Info   :Unhollower] 
[Info   :Unhollower] 439 fields restored
[Info   :Unhollower] 3 fields failed to restore
[Info   :Unhollower] Done in 00:00:00.0098765
[Info   :Unhollower] Unstripping methods...
[Info   :Unhollower] 
[Info   :Unhollower] 14862 methods restored
[Info   :Unhollower] 1046 methods failed to restore
[Info   :Unhollower] Done in 00:00:00.3088907
[Info   :Unhollower] Unstripping method bodies...
[Info   :Unhollower] 
[Info   :Unhollower] IL unstrip statistics: 7219 successful, 1899 failed
[Info   :Unhollower] Done in 00:00:00.2627747
[Info   :Unhollower] Generating forwarded types...
[Info   :Unhollower] Done in 00:00:00.0022732
[Info   :Unhollower] Writing xref cache...
[Info   :Unhollower] Done in 00:00:00.1941850
[Info   :Unhollower] Writing assemblies...
[Info   :Unhollower] Done in 00:00:00.9286795
[Info   :Unhollower] Writing method pointer map...
[Info   :Unhollower] Done in 00:00:00.0255553
[Info   :Unhollower] Done!
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Warning:Unhollower] Class::Init signatures have been exhausted, using il2cpp_class_has_references as a substitute!
`

**Comment by plasmatank** [comment]
Basically the same error
`
[LogOutput.log](https://github.com/BepInEx/BepInEx/files/8891852/LogOutput.log)
`

**Comment by guochaolinDLKF** [comment]
> 基本相同的错误
> 
> [消息：Preloader] BepInEx 6.0.0-be.568 - Touhou Blooming Soul [消息：Preloader] BLEEDING EDGE Build #568 from [e4c9ae4](https://github.com/BepInEx/BepInEx/commit/e4c9ae411502522e6495a601b02f1d872b4196c5) at master [Info: BepInEx] System platform: Windows 10 64-bit [Info: BepInEx] Process位数：64 位 (x64) [信息：Preloader] 在 Unity v2021.2.6 下运行 [消息：Preloader] 下载统一基础库 [消息：Preloader] 提取下载的统一基础库 [消息：Preloader] 生成 Il2CppUnhollower 程序集 [信息：Cpp2IL ] [库] 初始化元数据... [信息：Cpp2IL] [库] 使用实际的 IL2CPP 元数据版本 29 [信息：Cpp2IL] [库] 在 823 毫秒内初始化元数据 [Info : Cpp2IL] [Library] Searching Binary for required Data... [Info : Cpp2IL] [Library] 在 164 毫秒内得到二进制 codereg：0x181D2AF30，metareg：0x182048200。 [信息：Cpp2IL] [库] 正在初始化二进制文件... [信息：Cpp2IL] [库] 在 752 毫秒内初始化二进制文件 [信息：Cpp2IL] [库] 将指针映射到 Il2CppMethodDefinitions... [信息：Cpp2IL] [库] 已处理 90096 OK (103ms) [Info : Cpp2IL] [Program] Building assembly...这可能需要一些时间。 [信息：Cpp2IL] [程序] 在 1323 毫秒内完成构建组件 [信息：Cpp2IL] [程序] 修复显式覆盖。您在此处看到的任何警告都不是错误 - 它们通常表示不正确的剥离或混淆类型，但这没什么大不了的。这应该只需要一秒钟...... [信息：Cpp2IL] [程序]修复完成（207ms） [信息：Cpp2IL]总时间：00：00：03.5870177 [信息：预加载器]执行Il2CppUnhollower生成器 [信息：Unhollower]读取程序集... [Info :Unhollower] 于 00:00:00.0032346 完成 [Info :Unhollower] 读取系统程序集... [Info :Unhollower] 于 00:00:00.0228338 完成 [Info :Unhollower] 读取统一程序集... [信息：Unhollower] 完成于 00:00:00.0305602 [Info :Unhollower] 创建重写程序集... [Info :Unhollower] 完成于 00:00:00.0047722 [Info :Unhollower] 计算重命名... [Info :Unhollower] 完成于 00:00:00.0167373 [Info :Unhollower]创建 typedef... [Info :Unhollower] 完成于 00:00:00.0250811 [Info :Unhollower] 计算结构 blittability... [Info:Unhollower] 完成于 00:00:00.0022962 [Info :Unhollower] 填充 typedef... [Info :Unhollower] 于 00:00:00.0093291 完成 [Info :Unhollower] 填充通用约束... [Info :Unhollower] 于 00:00:00.0006570 完成 [Info :Unhollower] 创建成员... [Info :Unhollower] 00:00:00.2295572 完成 [Info :Unhollower] 扫描方法交叉引用... [Info :Unhollower] 完成于 00:00:00.3162036 [Info :Unhollower] 完成方法声明... [Info :Unhollower] 完成于 00:00:00.5870919 [Info :Unhollower] 总共 156 个可能死掉的方法 [Info :Unhollower] 填充方法参数... [Info :Unhollower] 完成于 00:00:00.1383634 [Info :Unhollower] 创建静态构造函数... [Info :Unhollower] 完成于 00 :00:00.1569565 [Info :Unhollower] 创建值类型字段... [Info :Unhollower] 完成于 00:00:00.0404219 [Info :Unhollower] 创建枚举... [Info :Unhollower] 完成于 00:00:00.0122509 [Info :Unhollower] 创建 IntPtr 构造函数... [Info :Unhollower] 完成于 00:00:00.0154403 [Info :Unhollower] 创建类型吸气剂... [Info :Unhollower] 完成于 00:00:00.0243499 [Info :Unhollower ] 创建非 blittable 结构构造函数... [Info :Unhollower] 完成于 00:00:00.0030089 [Info :Unhollower] 创建通用方法静态构造函数... [Info :Unhollower] 完成于 00:00:00.0155117 [Info : Unhollower] 创建字段访问器... [Info :Unhollower] 完成于 00:00:00.2879333 [Info :Unhollower] 填充方法... [Info :Unhollower] 完成于 00:00:00.7886715 [Info :Unhollower] 生成隐式转换... [Info :Unhollower] 于 00:00:00.0081187 完成 [Info :Unhollower] 创建属性... [Info :Unhollower] 于 00:00:00.0271470 完成 [Info :Unhollower] 取消剥离类型... [Info :Unhollower] 完成在 00:00:00.0162414 [Info:Unhollower] Unstripping fields... [Info:Unhollower] [Info:Unhollower] 439 个字段已恢复 [Info:Unhollower] 3 个字段未能恢复 [Info:Unhollower] 于 00:00 完成： 00.0098765 [Info :Unhollower] Unstripping 方法... [Info :Unhollower] [Info :Unhollower] 14862 方法已恢复 [Info :Unhollower] 1046 方法未能恢复 [Info :Unhollower] 完成于 00:00:00.3088907 [Info :Unhollower] Unstripping 方法体... [Info :Unhollower] [Info :Unhollower] IL unstrip 统计：7219 成功，1899 失败 [Info:Unhollower] 在 00:00:00.2627747 完成 [Info:Unhollower] 生成转发类型... [Info :Unhollower] 于 00:00:00.0022732 完成 [Info :Unhollower] 写入外部参照缓存... [Info :Unhollower] 于 00:00:00.1941850 完成 [Info :Unhollower] 写入程序集... [Info :Unhollower] 完成于 00:00:00.9286795 [Info:Unhollower] 写入方法指针映射... [Info:Unhollower] 完成于 00:00:00.0255553 [Info:Unhollower] 完成！ [信息：预加载器] 0 个补丁插件已加载 [信息：预加载器] 发现 0 个程序集 [消息：AssemblyPatcher] 正在执行 0 个补丁 [消息：BepInEx] Chainloader 已初始化 [警告：Unhollower] Class::Init 签名已用尽，使用 il2cpp_class_has_references 作为替代！

换成Unity2019.4版本就正常的，我的就是这样

**Comment by ghorsington** [comment]
Greetings!

Thanks for all the reports. Since we've been getting the same reports on Discord and other GH issues, I've consolidated all reports into a single ticket:

https://github.com/BepInEx/BepInEx/issues/474

You can find more information about the issue and a temporary workaround there. As such, I'm closing this issue to favour one central place for the problem discussion.

**Comment by Stellar-Lily** [comment]
I just came back to this issue and noticed it has been closed, but I believe issue #474 may not refer to the same issue as the one ALTDEUS showed at the time of opening? ALTDEUS runs on Unity 2019.4 and shows a different error.

I don't know enough about IL2CPP and BepInEx to know for sure if this is different, so please let me know if the replies (which do match the error and Unity versions of #474) and my error are covered by the new issue.
